data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -168,6 +164,24 @@ else if (isEventNameCreated(entry.getValue().get(0))) {
         return deleteMessageBatchRequestEntryCollection;
     }
 
+    private Optional<DeleteMessageBatchRequestEntry> processS3Object(
+            final Map.Entry<Message, List<S3EventNotification.S3EventNotificationRecord>> entry,
+            final S3ObjectReference s3ObjectReference) {
+        Optional<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntry = Optional.empty();
+        // SQS messages won't be deleted if we are unable to process S3Objects because of S3Exception: Access Denied
+        try {
```

Comment:
This try-catch block will suppress the exceptions thrown from #1544 . But, I'm still a little concerned that the current `while(true)` loop can have other exceptions thrown from it which result in this source mysteriously shutting down.

Reply:
I did check for any other cases of failure. Just to be sure, I will surround `processSqsMessages` in a try catch block.